src/dune_engine/targets.mli

```diff
@@ -63,3 +63,37 @@ val head : t -> Path.Build.t option
 val to_dyn : t -> Dyn.t
 
 val pp : t -> _ Pp.t
+
+(** The set of targets produced by an action. Each target may be tagged with a
+    payload, for example, the target's digest. *)
+module Produced : sig
+  type 'a t = private
+    { files : 'a Path.Build.Map.t
+    ; dirs : 'a Path.Build.Map.t Path.Build.Map.t
```

Comment:
`'a Path.Build.Map.t Path.Build.Map.t` feels a bit odd. I'd expect `'a String.Map.t Path.Build.Map.t`. Though, maybe it works better this way in the code.

Reply:
Let me try changing this to `'a String.Map.t Path.Build.Map.t`. I agree that this is better but it was a bit awkward in the first pass. Maybe now it's OK.

Reply:
OK, I made this change and I think the code got better. Thanks!